aws_lambda_builders/workflow.py

```diff
@@ -22,6 +25,40 @@
 Capability = namedtuple('Capability', ["language", "dependency_manager", "application_framework"])
 
 
+# TODO: Move sanitize out to its own class.
+def sanitize(func):
+    """
+    sanitize the executable path of the runtime specified by validating it.
+    :param func: Workflow's run method is sanitized
+    """
+
+    @functools.wraps(func)
+    def wrapper(self, *args, **kwargs):
+        valid_paths = []
+        # NOTE: we need to access binaries to get paths and resolvers, before validating.
+        self.get_binaries()
+        for binary_path in self.binaries:
+            validator = binary_path.validator
+            exec_paths = binary_path.resolver.exec_paths if not binary_path.path_provided else binary_path.binary_path
+            for executable_path in exec_paths:
+                valid_path = None
+                try:
+                    valid_path = validator.validate(executable_path)
+                except MisMatchRuntimeError as ex:
+                    LOG.debug("Invalid executable for %s at %s",
+                              binary_path.binary, executable_path, exc_info=str(ex))
+                if valid_path:
+                    binary_path.binary_path = valid_path
+                    valid_paths.append(valid_path)
+                    break
+        if not valid_paths:
```

Comment:
if `len(self.binaries())` is 2 but only 1 is a valid path, we will still execute the workflow, which will fail.